File: dynabuffers-java/src/test/kotlin/dynabuffers/usecase/Schema03Test.kt
package dynabuffers.usecase

import dynabuffers.AbstractDynabuffersTest
import dynabuffers.Dynabuffers
import org.junit.jupiter.api.Test

/**
 * @author benjamin.krenn@leftshift.one - 10/25/19.
 * @since 0.1.0
 */
class Schema03Test : AbstractDynabuffersTest() {

    @Test
    fun testParse() {
        val engine = Dynabuffers.parse(Schema03Test::class.java.getResourceAsStream("/schema03.dbs"))
        assertMap(engine, mapOf("results" to arrayOf<Map<String,Any>>(mapOf("text" to "hello world!"))))
    }

    @Test
    fun handlesEmptyArrays() {
        val engine = Dynabuffers.parse(Schema03Test::class.java.getResourceAsStream("/schema03.dbs"))
        assertMap(engine, mapOf("results" to emptyArray<Map<String,Any>>()))
    }
}

File: dynabuffers-python/tests/usecase/Schema03Test.py
import os
import unittest

from antlr4 import FileStream

from dynabuffers.Dynabuffers import Dynabuffers


class Schema03Test(unittest.TestCase):
    root_dir = os.path.dirname(os.path.realpath(__file__))

    def test_parse(self):
        engine = Dynabuffers.parse(FileStream(self.root_dir + "/schema03.dbs"))
        map = engine.deserialize(engine.serialize({"results": [{"text":"hello world"}]}))

        self.assertEqual(map, {"results": [{"text":"hello world"}]})


if __name__ == "__main__":
    unittest.main()
